File: php/tests/Doubles/InMemoryProductCatalog.php
<?php

namespace Tests\Pitchart\TellDontAskKata\Doubles;

use Doctrine\Common\Collections\ArrayCollection;
use Pitchart\TellDontAskKata\Domain\Product;
use Pitchart\TellDontAskKata\Repository\ProductCatalog;

class InMemoryProductCatalog implements ProductCatalog
{
    private ArrayCollection $products;

    /**
     * @param ArrayCollection $products
     */
    public function __construct(ArrayCollection $products)
    {
        $this->products = $products;
    }

    public function getByName(string $name): ?Product
    {
        return $this->products->findFirst(static function (int $key, Product $product) use ($name) {
            return $product->getName() === $name;
        });
    }
}

File: scala/src/test/scala/doubles/InMemoryProductCatalog.scala
package doubles

import ordershipping.domain
import ordershipping.repository.ProductCatalog

class InMemoryProductCatalog(val products: List[ordershipping.domain.Product])
    extends ProductCatalog {
  override def getByName(name: String): Option[domain.Product] =
    products.find(p => p.name == name)
}
